optimizely/decision_service.py

```diff
@@ -397,7 +435,7 @@ def get_variation_for_rollout(self, project_config, feature, user):
                                                                           bucketing_id)
                 decide_reasons.extend(bucket_reasons)
 
-                if bucketed_variation:
+                if isinstance(bucketed_variation, entities.Variation):
```

Comment:
This is a side effect of the dict[None, None] return type. The type checker doesn't know that the dict is always empty and thus that it would never end up in this branch of code. When I change the return type of `get_variation_from_id_by_experiment_id` and `get_variation_from_key_by_experiment_id` to `Optional[Variation]`, this check will no longer be necessary.